kotlin-spark-api/3.2/src/test/kotlin/org/jetbrains/kotlinx/spark/api/ApiTest.kt

```diff
@@ -593,6 +598,43 @@ class ApiTest : ShouldSpec({
                     it.nullable() shouldBe true
                 }
             }
+            should("Easily convert a (Java)RDD to a Dataset") {
+                // scala RDD
+                val rdd0: RDD<Int> = sc.parallelize(
+                    listOf(1, 2, 3, 4, 5, 6)
+                ).rdd()
+                val dataset0: Dataset<Int> = rdd0.toDS()
+                dataset0.show()
+
+                dataset0.toList<Int>() shouldBe listOf(1, 2, 3, 4, 5, 6)
+
+                // normal JavaRDD
+                val rdd1: JavaRDD<Int> = sc.parallelize(
+                    listOf(1, 2, 3, 4, 5, 6)
+                )
+                val dataset1: Dataset<Int> = rdd1.toDS()
+                dataset1.show()
+
+                dataset1.toList<Int>() shouldBe listOf(1, 2, 3, 4, 5, 6)
+
+                // JavaDoubleRDD
+                val rdd2: JavaDoubleRDD = sc.parallelizeDoubles(
+                    listOf(1.0, 2.0, 3.0, 4.0, 5.0, 6.0)
+                )
+                val dataset2: Dataset<Double> = rdd2.toDS()
+                dataset2.show()
+
+                dataset2.toList<Double>() shouldBe listOf(1.0, 2.0, 3.0, 4.0, 5.0, 6.0)
+
+                // JavaPairRDD
+                val rdd3: JavaPairRDD<Int, Double> = sc.parallelizePairs(
+                    listOf(Tuple2(1, 1.0), Tuple2(2, 2.0), Tuple2(3, 3.0))
+                )
+                val dataset3: Dataset<Tuple2<Int, Double>> = rdd3.toDS()
+                dataset3.show()
+
+                dataset3.toList<Tuple2<Int, Double>>() shouldBe listOf(Tuple2(1, 1.0), Tuple2(2, 2.0), Tuple2(3, 3.0))
```

Comment:
What about custom types like `c(1,2,"x")`?

Reply:
RDDs only work for classes that are serializable, similar to broadcasts. So unless we make the Arity's Serializable, no

Reply:
Are we sure there is no way to make non explicitly Serializable classes work here? If so — I think we should make arities serializable.

Reply:
`22/02/22 12:42:47 ERROR Utils: Exception encountered
java.io.NotSerializableException: org.jetbrains.kotlinx.spark.api.LonLat...`
is pretty clear. It needs a clsTag() for the scala api (and our `encoder().clsTag()` doesn't cut it). For the java api this is generated by scala on the fly. But in both cases the class needs to be serializable for the RDD to process it. So yes, I also think the Arities should be Serializable. Also makes them broadcastable :)

Reply:
Maybe there is a way to generate compatible `clsTag`?

Reply:
So, let's make'em serializable!

Reply:
I think just a clsTag won't do. it also needs a way to encode it, similar to Dataset. But in comparison to Dataset, just adding Serializable seems to fix it for all classes. However, Dataset is usually preferable for typed stuff anyways, so any compatibility with RDD is extra IMO.